src/linux_mcp_server/data_pipeline.py

```diff
@@ -0,0 +1,106 @@
+"""Classes and functions for ingesting and processing data."""
+
+import typing as t
+
+from mcp.server.fastmcp.exceptions import ToolError
+
+from linux_mcp_server.connection.ssh import execute_command
+from linux_mcp_server.utils.types import Host
+
+
+# Type aliases for the collect-parse-filter pattern
+CommandKey = tuple[str, ...]
+RawCommandOutput = str
+ParsedData = dict[str, t.Any]
+CommandList = list[list[str]]
```

Comment:
Is this a list that contains a list of strings or just a list of strings?

Reply:
It's a list of a list of strings. So that multiple command strings can be collected. For example:

```python
[
    ["cat", "/proc/cpuinfo"],
    ["cat", "/etc/os-release"]
]
```